app/assets/javascripts/application.js

```diff
@@ -19,5 +7,62 @@
 //= require gmaps
 
 document.addEventListener("turbolinks:load", function() {
-    initiateMap();
+  initiateMap();
+  getPosition({
+    enableHighAccuracy: true
+  }).then(function(position) {
+    var c = {
+      latitude: position.coords.latitude,
+      longitude: position.coords.longitude
+    };
+    setCookie('geocoderLocation', JSON.stringify(c)).then(function() {
+      redirectWithLocation();
+    }).catch(function(error) {
+      return console.log(error);
+    });
+  }).catch(function(error) {
+    return console.log(error);
+  });
 });
+
+getPosition = function getPosition(options) {
+  if (document.body.dataset.env === 'test') {
+    return new Promise(function(resolve) {
+      var fakePosition = JSON.parse(document.getElementById('fake_position').content);
+      resolve({
+        coords: {
+          latitude: fakePosition.coords.latitude,
+          longitude: fakePosition.coords.longitude
+        }
+      });
+    });
+  } else {
+    return new Promise(function(resolve, reject) {
+      navigator.geolocation.getCurrentPosition(function(position) {
+        return resolve(position);
+      }, function(error) {
+        return reject(error);
+      }, options);
+    });
+  }
+};
+
+function redirectWithLocation(position) {
+  var url = new URL(window.location.href);
+
+  if (document.body.dataset.geocoded != 'true') {
+    window.location.replace(url);
+    document.body.dataset.geocoded = true;
+  }
+}
+
+setCookie = function setCookie(name, value) {
```

Comment:
This repeat the function name. Please follow the standard set in other functions

Reply:
Done!